authority/tls.go

```diff
@@ -621,6 +633,18 @@ func (a *Authority) GetTLSCertificate() (*tls.Certificate, error) {
 	certTpl.NotBefore = now.Add(-1 * time.Minute)
 	certTpl.NotAfter = now.Add(24 * time.Hour)
 
+	// Policy and constraints require this fields to be set. At this moment they
+	// are only present in the extra extension.
+	certTpl.DNSNames = cr.DNSNames
+	certTpl.IPAddresses = cr.IPAddresses
+	certTpl.EmailAddresses = cr.EmailAddresses
+	certTpl.URIs = cr.URIs
+
+	// Fail if name constraints or policy does not allow the server names.
+	if err := a.isAllowedToSignX509Certificate(certTpl); err != nil {
+		return fatal(err)
+	}
```

Comment:
If I understand correctly, signing the CA leaf certificate will now only work if the constraints and the policy allow it to. I seem to have missed this case when implementing allow/deny, either intentionally or unintentionally; I don't remember that.

Does it make sense to only check the name constraints here, like on renew? If we would start blocking leaf certificates from being issued because of policy, then users could have a hard time if they didn't set it correctly in the first place. They may be able to fix it, but I can imagine cases in which it's harder for them to do so. I think we can also make the case that for this specific flow for getting a leaf certificate we want to keep it easy for users and we're in control of what's being signed exactly at what time, so not enforcing the policy in this specific case seems acceptable to me. This may have been my reasoning at time of implementation too.

Reply:
I will only use name constraints here. The policy can be used to avoid generating a new certificate with the CA names.

Reply:
Fixed with c9e7af37